crates/resolver/src/dns_lru.rs

```diff
@@ -42,6 +42,32 @@ impl LruValue {
     fn ttl(&self, now: Instant) -> Duration {
         self.valid_until.saturating_duration_since(now)
     }
+
+    fn with_updated_ttl(&self, now: Instant) -> Self {
+        let lookup = match self.lookup {
+            Ok(ref lookup) => {
+                let records = lookup
+                    .records()
+                    .iter()
+                    .map(|record| {
+                        let mut record = record.clone();
```

Comment:
While this clone is correct, it could be expensive, which is unfortunate. I’m not sure I have a better solution though.

Reply:
It looks like we were cloning anyway. So this isn’t a new introduction.